### PR TITLE
Initial webgl cleanup

### PR DESCRIPTION
**Pull request by bryevdv**
This PR:

* re-integrates the webgl support back in to the main bundle
* removes some dead/unused code
* provides and uses an API function to determine uniform/single-value-ness rather than digging in property internals
* affords configuration of markers GL support at the point marker classes are created
* consolidates all marker-specific GL code not used elsewhere in to `webgl/markers.ts` 

@mattpap This is some prep work. With this out of the way I think work can proceed almost entirely in `webgl/markers.ts` (and later in corresponding `line.ts`) with the addition of a few lines in `canvas.ts` as well. Hopefully this will be incremental and non-disruptive enough for you. Assuming this can get green soon, I'd be OK merging this prep work for 2.2 but also fine landing on `branch-2.3`. Let me know your thoughts, or any other suggestions for improvements. 

incidentally I also have figured out distance-field fragment shaders for several other marker types that are currently missing but I would plan to add those separately later.

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap any thoughts on the failing integration test? My offhand recollection is that recently the circle GL examples were actually falling back to canvas. Perhaps now they are not and there is a discrepancy? I will note I also discovered that the `scatter10k.py` example works, but *only* if you pass an explicit list of colors. Otherwise there is an out of bounds error, at least on Safari.

**Comment by bryevdv** [comment]
@mattpap  I guess this was related to the Chrome issue PR you just made? After merging the latest changes all but one (actually webgl-related) test are passing. Would still appreciate your input on when to land this.

File: bokehjs/src/lib/models/markers/marker.ts [hunk 1]
```diff
@@ -25,10 +27,20 @@ export abstract class MarkerView extends XYGlyphView {
   visuals: Marker.Visuals
 
   /** @internal */
-  glglyph?: MarkerGLGlyph
+  glglyph_cls?: Class<MarkerGL>
```
Comment (bryevdv) [inline bokehjs/src/lib/models/markers/marker.ts:30]: `glglyph_cls` is temporary but the point is that GL-enablement for markers happens explicitly in the one place where the markers are defined, not implicitly in multiple parts of the codebase.

**Comment by bryevdv** [comment]
Would really like to have the `eslint` job be included in the `codebase` tests, even if it runs twice, separately in BokehJS-CI

@mattpap Without comment, I will plan to merge this in `branch-2.2` EOD tomorrow PST.

File: bokehjs/src/lib/models/markers/marker.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@ import * as p from "core/properties"
 import {range} from "core/util/array"
 import {Context2d} from "core/util/canvas"
 import {Selection} from "../selections/selection"
+import {Class} from "core/class"
+
```
Comment (mattpap) [inline bokehjs/src/lib/models/markers/marker.ts:14]: Unnecessary line.

File: bokehjs/test/codebase/size.ts [hunk 3]
```diff
@@ -8,17 +8,15 @@ const build_dir = path.normalize(`${__dirname}/../..`) // build/test/codebase ->
...
   // es2017
-  ["js/bokeh.min.js",                700],
-  ["js/bokeh-widgets.min.js",        300],
-  ["js/bokeh-tables.min.js",         270],
-  ["js/bokeh-api.min.js",             90],
-  ["js/bokeh-gl.min.js",              70],
+  ["js/bokeh.min.js",               750],
+  ["js/bokeh-widgets.min.js",       300],
+  ["js/bokeh-tables.min.js",        270],
+  ["js/bokeh-api.min.js",            90],
```
Comment (mattpap) [inline bokehjs/test/codebase/size.ts:14]: Alignment is off.

File: bokehjs/src/lib/models/markers/defs.ts [hunk 4]
```diff
@@ -447,31 +449,31 @@ function _mk_model(type: string, f: RenderOne): Class<Marker> {
...
-export const Square           = _mk_model('Square',           square)
-export const SquareCross      = _mk_model('SquareCross',      square_cross)
-export const SquareDot        = _mk_model('SquareDot',        square_dot)
-export const SquarePin        = _mk_model('SquarePin',        square_pin)
-export const SquareX          = _mk_model('SquareX',          square_x)
-export const Triangle         = _mk_model('Triangle',         triangle)
-export const TriangleDot      = _mk_model('TriangleDot',      triangle_dot)
-export const TrianglePin      = _mk_model('TrianglePin',      triangle_pin)
-export const X                = _mk_model('X',                x)
-export const Y                = _mk_model('Y',                y)
```
Comment (mattpap) [inline bokehjs/src/lib/models/markers/defs.ts:474]: Unnecessary changes; should have added only the new arguments.

File: bokehjs/make/paths.ts [hunk 5]
```diff
@@ -32,10 +32,6 @@ export const lib = {
     main: join(build_dir.lib, "main.js"),
     output: join(build_dir.js, "bokeh.js"),
   },
-  gl: {
-    main: join(build_dir.lib, "models/glyphs/webgl/main.js"),
-    output: join(build_dir.js, "bokeh-gl.js"),
-  },
```
Comment (mattpap) [inline bokehjs/make/paths.ts:38]: This is a breaking change, especially for anyone who uses bokehjs directly, so for backwards compatibility you have to keep an empty bundle until 3.0, similarly to how I kept `*.css` bundles pre 2.0.

File: bokehjs/src/lib/models/markers/defs.ts (hunk 4 above)
Comment (bryevdv) [reply bokehjs/src/lib/models/markers/defs.ts:474]: I consider them advised. The alignment is a pain to maintain, and with the additional argument, then lines become excessively long. In general I would like to avoid (and undo) forced alignments on this scale.

File: bokehjs/test/codebase/size.ts (hunk 3 above)
Comment (bryevdv) [reply bokehjs/test/codebase/size.ts:14]: If there is going to be alignment at all, it should encompass both the "top" and "bottom" halves.

File: bokehjs/make/paths.ts (hunk 5 above)
Comment (bryevdv) [reply bokehjs/make/paths.ts:38]: We could handle publishing an empty file in the release automation. I think that is the simplest option that only requires (future undo-able) changes in one place and keeps the main codebase cleaner. I will proceed that way later today unless you have other considerations to offer.

Comment (bryevdv) [reply bokehjs/make/paths.ts:38]: Though I would also point out: If anyone is manually including BokehJS  bundles, then updating/changing versions at all will already require manual interventions, so I am not quite sure why another manual intervention to delete a bundle is any different from the user POV.

Comment (mattpap) [reply bokehjs/make/paths.ts:38]: > If anyone is manually including BokehJS bundles, then updating/changing versions at all will already require manual interventions, (...)

> We could handle publishing an empty file in the release automation.

There's an option of using bundles available via bokehjs' npm package, and those don't have version information. 

In general I'm not sure if care that much and the simpler the build the better. We can reconsider this before the release.

Comment (bryevdv) [reply bokehjs/make/paths.ts:38]: > We can reconsider this before the release.

@mattpap Just to make sure I understand, you are agreeing that this can be merged now, as is?
